extensions/ql-vscode/src/vscode-tests/ensureCli.ts

```diff
@@ -105,6 +106,15 @@ export async function ensureCli(useCli: boolean) {
   }
 }
 
+/**
+ * Heuristically determines if the codeql libraries are installed in this
+ * workspace. Looks for the existance of a folder whose path ends in `/codeql`
+ */
+export function hasCodeQL() {
+  const folders = workspace.workspaceFolders;
+  return !!folders?.some(folder => folder.uri.path.endsWith('/codeql'));
```

Comment:
A more principled way would be to run `codeql resolve qlpacks` and make sure we've picked up the packs in the `codeql` repo. I think that is worth doing as an integration test, but perhaps not necessary here just to verify your test setup.

Reply:
Sure. I'll add that.